src/main/java/com/kenshoo/swagger/validator/MimeValidator.java

```diff
@@ -0,0 +1,275 @@
+     *
+     * @param cls class to look at
+     * @param httpMethod jax-rs annotation
+     *
+     * @return Method
+     */
+    private Method getJaxRsMethod(Class<?> cls, String httpMethod) {
+        Class<? extends Annotation> annotationClass = resolveJaxRsAnnotation(httpMethod);
+        for (Method m : cls.getMethods()) {
+            Object jaxrsMethodAnnotation = m.getAnnotation(annotationClass);
+            Object pathAnnotation = m.getAnnotation(Path.class);
+            if (jaxrsMethodAnnotation != null && pathAnnotation == null){
+                return m;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Locates corresponding jax-rs annotation by it's name
+     *
+     * @param httpMethod method name
+     *
+     * @return jax-rs annotation
+     */
+    private Class<? extends Annotation> resolveJaxRsAnnotation(String httpMethod) {
+        if(!Strings.isNullOrEmpty(httpMethod)) {
+            switch (httpMethod) {
+                case "get":
+                    return GET.class;
+                case "post":
+                    return POST.class;
+                case "put":
+                    return PUT.class;
+                case "delete":
+                    return DELETE.class;
+
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Validates that mime annotation exist either on Class or on Method
+     *
+     * @param cls
+     */
+    private void validateMimeAnnotationsExistence(Class<?> cls) {
+        Produces producesMethodAnnotation, producesClassAnnotation = cls.getAnnotation(Produces.class);
+        Consumes consumesMethodAnnotation, consumesClassAnnotation = cls.getAnnotation(Consumes.class);
+
+        for (Method m : cls.getMethods()) {
+            if (isJaxRsMethod(m)) {
+                producesMethodAnnotation = m.getAnnotation(Produces.class);
+                if (producesClassAnnotation == null && producesMethodAnnotation == null) {
+                    messageEventHandler.handleError(
+                            "Produces annotation is not found neither " +
+                            "on {0} level nor on method {1}.", cls, m
+                    );
+                }
+                consumesMethodAnnotation = m.getAnnotation(Consumes.class);
+                if (consumesClassAnnotation == null && consumesMethodAnnotation == null) {
+                    messageEventHandler.handleError(
+                            "Consumes annotation is not found neither " +
+                            "on {0} level nor on method {1}.", cls, m
+                    );
+                }
+            }
+        }
+    }
+
+    /**
+     * Checks whether given method has one of
+     * get/post/put/delete jax-rs annotations
+     *
+     * @param m method to check
+     *
+     * @return boolean
+     */
+    private boolean isJaxRsMethod(Method m) {
+        return m.getAnnotation(GET.class) != null   ||
```

Comment:
You need to check if method is annotated with annotation of HttpMethod. You may have other methods (HEAD, OPTIONS, PATCH, etc.)
See the implementation of ResourceValidator.isOperationAnnotatedMethodExists
Maybe you can also reuse some code.